lib/create-plugin.js

```diff
@@ -62,7 +70,16 @@ function createPlugin(instrumentationApi) {
                 longestResolvedPath.formatted = formattedPath
               }
 
-              return () => {
+              return (error) => {
+                if (error) {
+                  // This handler is invoked prior to didEncounterErrors
+                  // which means we need to handle the error now to capture
+                  // in context of the appropriate span.
```

Comment:
Oh, ha, this comment kind of answers my question from above. But if the didEncounterErrors code does the work to getting the appropriate context, why is this necessary? Also, this seems to push to NOTICED_ERRORS on the request context, but the didEncounterErrors code path doesn't. Does the error get sent here first, and then to didEncounterErrors, and so that code path checks for duplicates?

Reply:
Ya it looks like this calls noticeError directly and didEncounterErrors will check for noticed errors and then call noticeError.

Reply:
Yeah. So this captures the errors against the appropriate resolve field segment/span. Then flags as noticed so we don't re-capture with the `didEncounterErrors` handler, which would associate to the operation segment/span.